dask/array/linalg.py

```diff
@@ -157,10 +158,13 @@ def tsqr(data, name=None, compute_svd=False):
 
     if not compute_svd:
         qq, rr = np.linalg.qr(np.ones(shape=(1, 1), dtype=data.dtype))
+        q_shape = data.shape if data.shape[0] >= data.shape[1] else (data.shape[0], data.shape[0])
+        q_chunks = data.chunks if data.shape[0] >= data.shape[1] else (data.chunks[0], data.chunks[0])
+        r_shape = (n, n) if data.shape[0] >= data.shape[1] else data.shape
         q = Array(dsk, name_q_st3,
-                  shape=data.shape, chunks=data.chunks, dtype=qq.dtype)
+                  shape=q_shape, chunks=q_chunks, dtype=qq.dtype)
         r = Array(dsk, name_r_st2,
-                  shape=(n, n), chunks=(n, n), dtype=rr.dtype)
+                  shape=r_shape, chunks=n, dtype=rr.dtype)
```

Comment:
Is this change tested anywhere?  

Alternatively, could this change be moved into a separate PR?  It makes it harder to merge the sfqr changes (which seem good to go) when they become attached to new changes that also need review.

Reply:
They are exercised in the new tests.

In fact, it fixes the bug reported in: https://github.com/dask/dask/issues/3583
```py
a = np.random.random((3, 5))
data = da.from_array(a, chunks=5)
q, r = da.linalg.tsqr(data)
q.shape, r.shape
```
we get...
```py
((3, 5), (5, 5))
```

Reply:
```
q.compute() # yields a 3x3 matrix
```

Reply:
Yes, but let me go make sure that the shapes are explicitly checked.

Reply:
Just added some explicit tests.